SquareLine/README.md

```diff
@@ -0,0 +1,56 @@
+# SquareLine board generator
+
+This is simple python generator for SquareLine board packages. This generator copy all files from `common` folder to destination ZIP file and generates other files by `manifest.json` file saved in each board directory in `boards` folder. 
+
+## Mandatory files in each board folder:
+* `manifest.json` - Description of the board
+* `image.png` - Image of the board 380px x 300px
+* `sdkconfig.defaults` - defaults
+* `main/idf_component.yml` - Component Registry description
+
+### `manifest.json`
+
+Mandatory JSON keys in manifest:
+* `name` - Board name (string)
+* `version` - Package version (string)
+* `mcu` - ESP mcu (string)
+* `screen_width` - Screen width (string)
+* `screen_height` - Screen height (string)
+* `screen_color_swap` - Is color bytes swapped (boolean)
+* `short_description` - Short board description (string)
+* `long_description` - Long board description (string)
+* `placeholders` - List of placeholders and values of them (JSON object)
+
+### Placeholders
+
+During the copy all files (except `image.png`), there can be replaced some placeholder by the value. Placeholder must be in compound brackets (e.g. `{PLACEHOLDER}`). The values for placeholders are saved in `manifest.json` file without compound brackets like this:
+
+```
+    "placeholders":
+    {
+        "PLACEHOLDER_1": "value 1",
+        "PLACEHOLDER_2": "value 2"
+    }
+```
+
+**_NOTE 1:_** Placeholders are optional.
```

Comment:
I'm not sure what "optional" means here. The note below says that if the placeholder is not defined in Json, it will not be replaced in the source file. So it looks like it's not optional?

Reply:
It means, that all JSON keys must be there except placeholders. When there is missing some other key, the script fails. When there is missing some placeholder, it continues, but not replace the placeholder.